### PR TITLE
Add content to jsondb.md

### LINKED ISSUE #202: JsonDB storage description
http://docs.openhab.org/administration/jsondb.html

A short introduction into jsondb would be enough. The process of how a user could (if ever needed) edit the db file manually would be good too. Most info can probably be taken from the community forum thread https://community.openhab.org/t/replacement-of-mapdb-by-jsondb/17498

@cdjackson would you take this on?

### PR DESCRIPTION
**Pull request by AngelosF**
Closes #202 (https://github.com/openhab/openhab-docs/issues/202)

Sources of info:
a) https://community.openhab.org/t/replacement-of-mapdb-by-jsondb/17498
b) https://github.com/eclipse/smarthome/pull/1868

Signed-off-by: Angelos <agf@itcs.ro> (github: AngelosF)

### CODE REVIEW

**Comment by ThomDietrich** [comment]
Just to leave a note for others: This PR is work in progress.

**Comment by AngelosF** [comment]
Yes, sorry for the delay... I just couldn't find enough time to progress it
I should be able to spend some time on it today

**Comment by ThomDietrich** [comment]
No worries! Take your time ;) Thanks for your efforts!

**Comment by ThomDietrich** [comment]
A friendly poke ;)

**Comment by bgilmer77** [comment]
My two cents - this is an EXTREMELY useful contribution.  Any chance we can publish it as it stands?  It would have been really valuable to have had access to this as I was transitioning from OH1 to OH2.  My first reaction after starting OH2 was, "Hey!  Where'd all my configuration stuff go?"  followed almost immediately by, Rats - I bet it is now buried deep inside some binary database somewhere - this sucks :-(

I am happy to contribute to finishing this up if y'all (a distinctly Southern American grammatical construction) need help.  (FWIW, I don't speak "Southern" American, but I can interpret for you if you ever need it ;-)

**Comment by ThomDietrich** [comment]
@AndrewMorsillo I've updated the PR to include the latest content. It now consists of two independent parts that need to merged properly.

**Comment by Confectrician** [comment]
@bgilmer77 could you have a look at this improvements from a reading side?

I have checked the current jsondb page and we are not conflicting with the existing content, but add some valuable content at the end. (Even if it is WIP for now.)

@cdjackson could you maybe have a short look too?

I would like to merge this PR in the nearer future, even if it was inactive for some time now.
If there are some small improvements/fixes to do, i would provide them by myself.

@ThomDietrich could you maybe also give a statement here?

File: administration/jsondb.md [hunk 1]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
 
 ```
+
+----
+
+openHAB stores configuration information in JSON (JavaScript Object Notation) formatted (structured) text files located in the `OPENHAB_USERDATA/jsondb/` directory.
+
+## Storage Scope
+
+All configuration information regarding _**Items, Links, and Things**_ which are defined via the User Interfaces (Paper UI, HABmin, REST) or via internal openHAB services.
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider, "regarding _**Items, Links, and Things**_ are defined via..."

File: administration/jsondb.md [hunk 2]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
+
+Note: The JSON DB does NOT store information for manually configured Items, Links, or Things, since these are already stored in files within the `OPENHAB_CONF` sub-directories (e.g. `/etc/openhab2/items/`).
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider formatting "Items, Links, or Things" the same way as above?

File: administration/jsondb.md [hunk 3]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
+
+## Storage Purpose
+
+The JSON DB Storage can be used for:
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider eliminating the word "The" at the beginning of, "The JSON DB Storage..."

File: administration/jsondb.md [hunk 4]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
+- Backup: Maintains a copy of your configurations in the `OPENHAB_USERDATA/jsondb/backup` directory
+- Troubleshooting: Allows the user to interact with the configurations that were automatically generated via the UIs
+- Advanced administration: Gives the possibility to manually define configuration parameters within the `*.json` files
+
+## Storage Use
+
+openHAB writes the `*.json` files everytime a change is made via the User Interfaces.
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider, "files every time a change..."

File: administration/jsondb.md [hunk 5]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
+openHAB _**reads only once the `*.json` files at startup**_.  So, if you edit them manually, you should restart openHAB.
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider, "+openHAB _**reads the `*json` files only once at startup**_

File: administration/jsondb.md [hunk 6]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
+
+The system employs 2 write mechanisms  to improve performance where there are multiple writes in a short time. When the service is closed, all open services are written to disk.  The parameters for the 2 mechanisms can be modified in Paper UI :arrow_right: Configuration :arrow_right: System :arrow_right: Json Storage
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider, "The system employs two write mechanisms..."

Eliminate extra space between "mechanisms" and "to improve"

Consider, "The parameters for the two mechanisms may be modified in the Paper UI..."

File: administration/jsondb.md [hunk 7]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
+
+1. _Write delay_ (defaults to 500 ms): Sets the time to wait before writing changes to disk. This can reduce the number of writes when many changes are being introduced within a short period, and
+2. _Maximum write delay_ (defaults to 30000 ms): Sets the maximum period the service will wait to write data to disk in the event that many changes are happening continually.
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider, "will wait to write data in cases where changes are happening continually."

File: administration/jsondb.md (hunk 6 above)
Comment (bgilmer77) [inline administration/jsondb.md]: I think the linter would say one sentence per line unless this is in a bulleted list.  Please put each sentence on its own line.  This comment applies to other areas of this PR as well.

File: administration/jsondb.md [hunk 8]
```diff
@@ -30,3 +30,154 @@ Example file -:
...
+
+The service keeps backups of each table - currently up to 5 backups are maintained. Each time the json file is written, the old file is moved into a backup folder, and old files are removed from the backup folder.
```
Comment (bgilmer77) [inline administration/jsondb.md]: Consider, "Each time a new JSON file is written, the old file..."

Comment (Confectrician) [reply administration/jsondb.md]: I would disagree here.
We don't write a new file each time, we just overwrite and backup the (now) outdated version.

Unfortunately i  don't know how to write it better.
Maybe we should leave it this way for now.

File: administration/jsondb.md [hunk 9]
```diff
@@ -30,3 +45,158 @@ Example file -:
 ```
 
 ```
```
Comment (Confectrician) [inline administration/jsondb.md]: Do we need this empty example file code block or does the new content make this obsolete?

File: administration/jsondb.md (hunk 8 above)
Comment (bgilmer77) [reply administration/jsondb.md]: Got it - how about, "The service keeps up to five backups of each table.  The outdated file is copied to the backup folder and then that file is overwritten with the new file."

Comment (Confectrician) [reply administration/jsondb.md]: Sounds perfect. I will edit it in this way.

**Comment by Confectrician** [comment]
@bgilmer77 wdyt, can we merge this?

One thing left would be to move the pictures from imageshack to our repo,
but i want to do this after merge in another pr.

**Comment by bgilmer77** [comment]
I see a few things that I would comment on but I cannot do it until next week.  If that is too late, go ahead and merge.  Otherwise I will submit edits when I can.

**Comment by Confectrician** [comment]
Friendly poke to @bgilmer77 :)

**Comment by Confectrician** [comment]
I will wait till sunday and then merge this.
It is an improvement anyway and we can continue improving it afterwards.

It doesn't make sence to keep that as a PR so close to a fnished state.

**Pull request merged**